website/components/grid/Grid.tsx

```diff
@@ -59,23 +52,22 @@ type GridContainerProps = JSX.IntrinsicElements['div'] & {
   wrap?: Wrap;
 };
 
-export function GridContainer({
-  direction = Direction.Row,
-  align = Align.Center,
-  justify = Justify.Center,
-  wrap = Wrap.Wrap,
-  children,
-}: GridContainerProps) {
-  const gridContainerStyles = css`
-    ${baseContainerStyles};
-    flex-direction: ${direction};
-    flex-wrap: ${wrap};
-    align-items: ${align};
-    justify-content: ${justify};
-  `;
-
-  return <div css={gridContainerStyles}>{children}</div>;
-}
+const GridContainer = styled<'div', GridContainerProps>('div')`
+  ${mq({
+    width: breakpointGaps.map(gap => `calc(100% + ${gap})`),
+    margin: breakpointGaps.map(gap => `calc(-1 * ${gap} / 2)`),
+  })}
+
+  display: flex;
+  ${['flex-direction']}: ${props =>
+    props.direction ? props.direction : Direction.Row};
+  ${['flex-wrap']}: ${props => (props.wrap ? props.wrap : Wrap.Wrap)};
+  ${['align-items']}: ${props => (props.align ? props.align : Align.Center)};
+  ${['justify-content']}: ${props =>
+    props.justify ? props.justify : Justify.Center};
+`;
```

Comment:
not a big fan of this api. it looks more like its own language than javascript and is significantly harder to read than the previous code.

Reply:
Updated because @DesignerDave loves styled components. Let me know what you think in this case!

Reply:
I'd have to hear a real rationale behind the preference. the readability of this code does not make a very convincing case.